code/game/turfs/turf.dm

```diff
@@ -34,26 +34,22 @@
 	var/can_build_into_floor = FALSE // Used for things like RCDs (and maybe lattices/floor tiles in the future), to see if a floor should replace it.
 	var/list/dangerous_objects // List of 'dangerous' objs that the turf holds that can cause something bad to happen when stepped on, used for AI mobs.
 
-/turf/New()
-	..()
-	for(var/atom/movable/AM as mob|obj in src)
-		spawn( 0 )
-			src.Entered(AM)
-			return
-	turfs |= src
-
-	if(dynamic_lighting)
-		luminosity = 0
-	else
-		luminosity = 1
-
+/turf/Initialize(mapload)
+	. = ..()
+	for(var/atom/movable/AM in src)
+		Entered(AM)
+
+	//Lighting related
+	luminosity = !(dynamic_lighting)
+	has_opaque_atom = (opacity)
```

Comment:
This will set overwrite true values with false if turf is not opaque, even if it has an opaque atom.
```suggestion
	has_opaque_atom |= (opacity)
```

Reply:
Did this